protocol/transport.go

```diff
@@ -31,40 +45,59 @@ func (t *Transport) endTransaction() (err error) {
 	return
 }
 
+func (t *Transport) affectTransaction(msg pgproto3.FrontendMessage) (ts TransactionState, err error) {
+	if t.transaction == nil {
+		switch msg.(type) {
+		case *pgproto3.Parse, *pgproto3.Bind, *pgproto3.Describe:
+			t.beginTransaction()
+			ts = InTransaction
+		default:
+			ts = NotInTransaction
+		}
+	} else {
+		if t.transaction.hasError() {
+			ts = TransactionFailed
+		}
+		switch msg.(type) {
+		case *pgproto3.Query, *pgproto3.Sync:
+			err = t.endTransaction()
+			if err != nil {
+				ts = TransactionFailed
+			} else if ts == 0 {
```

Comment:
Please use a const instead of `0`. Same for the `default` case. I think it will require to add `TransactionUnknown` value.

Do you have a better option to avoid using magic numbers and exposing internal `TransactionState` type?

Reply:
added `TransactionUnknown`